src/goImport.ts

```diff
@@ -13,9 +13,9 @@ import { getImportablePackages } from './goPackages';
 
 const missingToolMsg = 'Missing tool: ';
 
-export function listPackages(excludeImportedPkgs: boolean = false): Thenable<string[]> {
+export function listPackages(excludeImportedPkgs: boolean = false, workDir?: string): Thenable<string[]> {
```

Comment:
why is this change needed when `getImportablePackages` can calculate the `workDir` by using the given `fileName`?

Reply:
I need this to pass the test of **"Replace vendor packages with relative path"** it will fail on the assert of *"Relative path for vendor package 9fans.net/go/acme not found"*.

This happen because `gopkgs` recognize vscode workspace as `workDir`, and the workspace dir for the running test is `vscode-go/out/test`, no vendor package under the directory. While actually we are testing vendor package on under `$GOPATH/src/github.com/rogpeppe/godef`, there the vendor package of `9fans.net/go/acme` should be found.

So it only for satisfying the test. The actual running vscode will never use `workDir` param or `listPackages()` (which kind of silly). I can't guess the project root/`workDir` based on  `fileName`. Any suggestion?

Available the options are:
1. Guess golang project dir by filename
2. Change the test workspace dir to `$GOPATH/src/github.com/rogpeppe/godef` above test
3. Pass `workDir` to `listPackages()` (currently use this)

Reply:
@uudashr This can happen IRL too. A user might open and edit a Go file that is not under the current workspace.